streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -162,11 +169,11 @@ private[streaming] class ReceiverSupervisorImpl(
   }
 
   override protected def onStart() {
-    blockGenerator.start()
+    registeredBlockGenerators.foreach { _.start() }
```

Comment:
Since this line is called before `startReceiver()`, `BlockGenerator` in `ReliableKafkaReceiver` should be started and stopped by the receiver itself. Right?